library/alloc/src/vec/mod.rs

```diff
@@ -2874,13 +2874,12 @@ impl<T, A: Allocator> Vec<T, A> {
             );
             self.reserve(additional);
             unsafe {
-                let mut ptr = self.as_mut_ptr().add(self.len());
+                let ptr = self.as_mut_ptr();
                 let mut local_len = SetLenOnDrop::new(&mut self.len);
                 iterator.for_each(move |element| {
-                    ptr::write(ptr, element);
-                    ptr = ptr.add(1);
-                    // Since the loop executes user code which can panic we have to bump the pointer
-                    // after each step.
+                    ptr::write(ptr.add(local_len.current_len()), element);
```

Comment:
This looks pointless, but seems like it reduces register pressure since we have to maintain the `local_len` for panic safety anyway.

Before:
```nasm
.LBB4_3:
	mov	qword ptr [rbp - 16], rax
	mov	qword ptr [rbp - 24], rcx    ; Note Spill
	call	make_thing
	mov	rcx, qword ptr [rbp - 24]
	mov	dword ptr [rcx], eax
	add	rcx, 4
	mov	rax, qword ptr [rbp - 16]
	dec	rax
	cmp	rsi, rax
	jne	.LBB4_3
```

After:
```nasm
.LBB6_3:
	mov	qword ptr [rbp - 16], rax
	call	make_thing
	mov	rdx, qword ptr [rbp - 16]
	lea	rcx, [rdx + 1]
	mov	dword ptr [rbx + 4*rdx], eax    ; Note addressing mode
	mov	rax, rcx
	dec	rsi
	jne	.LBB6_3
```

(And LLVM understands that kind of loop *very* well too, since it's a `for i in A..B { v[i] = foo(); }` loop that's common all over the place.)